packages/material-ui/src/InputBase/InputBase.js

```diff
@@ -229,6 +230,15 @@ class InputBase extends React.Component {
   handleRefInput = ref => {
     this.inputRef = ref;
 
+    warning(
+      !ref || ref instanceof HTMLElement || ref.focus,
```

Comment:
If we only need the ref for focus we might as well just interface it as 
```ts
interface InputRef {
  focus(): void
}
```
for which `typeof ref.focus === 'function'` is better. Right now this could still allow non-callable truthy values for `ref.focus`. This is mainly aimed at function components in react 16.7. You could pass your own input function component that uses [`useImperativeMethods`](https://reactjs.org/docs/hooks-reference.html#useimperativemethods) to interface focus managment.
Unless we need more from `HTMLElement`. 

For readability I would prefer this at the start of the method body (especially before setting our instance variable).

Reply:
We also look for `.value` and `.defaultValue`. I was proposing the usage of `ref.focus` as a "hack" to comply with the select component. I'm happy to improve the logic by using a function typeof and an exhaustive check.

Reply:
So it is actually an `HTMLInputElement` not just any `HTMLElement`. Then the instanceof should cover everything.

Reply:
Using `HTMLInputElement` sounds better 👍. The select returns a "mock", I believe something similar to what `useImperativeMethods` would return.

Reply:
That's what I though. I think we can use a simplified duck type test for `.focus` here and document that we expect the ref to implement the `HTMLInputElement` interface.